docker/test/fuzzer/run-fuzzer.sh

```diff
@@ -173,16 +173,15 @@ function fuzz
 
     mkdir -p /var/run/clickhouse-server
 
-    # NOTE: we use process substitution here to preserve keep $! as a pid of clickhouse-server
-    # server.log -> CH logs
-    # stderr.log -> Process logs (sanitizer)
+    # server.log -> All server logs, including sanitizer
+    # stderr.log -> Process logs (sanitizer) only
     clickhouse-server \
         --config-file db/config.xml \
         --pid-file /var/run/clickhouse-server/clickhouse-server.pid \
         --  --path db \
             --logger.console=0 \
-            --logger.log=server.log > stderr.log 2>&1 &
-    server_pid=$!
+            --logger.log=server.log 2>&1 | tee -a stderr.log >> server.log 2>&1 &
+    server_pid=$(pidof clickhouse-server)
```

Comment:
`$!` looks a bit safer to me, because it return pid of just started bg process, not looking into process tables

Reply:
`$!` will return the pid of `tee`

```
/tmp $ /mnt/ch/ClickHouse/build_msan/programs/clickhouse server --pid-file /tmp/server.pid -- --logger.console=0 --logger.log=server.log 2>&1 | tee -a stderr.log >> server.log 2>&1 &
[2] 153446
[1]   Done                    /mnt/ch/ClickHouse/build_msan/programs/clickhouse server --pid-file /tmp/server.pid -- --logger.console=0 --logger.log=server.log 2>&1 | tee -a stderr.log >> server.log 2>&1
/tmp $ echo $!
153446
/tmp $ ps aux | grep 153446
raul      153446  0.0  0.0   8012  1792 pts/4    S    14:38   0:00 tee -a stderr.log
raul      154168  0.0  0.0   9092  2560 pts/4    S+   14:38   0:00 grep --text 153446
```

Reply:
Indeed